lib/Frontend/ModuleInterfaceLoader.cpp

```diff
@@ -1514,6 +1514,17 @@ std::error_code ExplicitSwiftModuleLoader::findModuleFilesInDirectory(
   return std::error_code();
 }
 
+bool ExplicitSwiftModuleLoader::canImportModule(
+    Located<Identifier> mID) {
+  StringRef moduleName = mID.Item.str();
+  auto it = Impl.ExplicitModuleMap.find(moduleName);
```

Comment:
This is not something you would change in this PR, but…why is `ExplicitModuleMap` keyed by `StringRef`s? I would expect `Identifier` keys to be faster, since `Identifier`s are interned and can be compared more efficiently.

Reply:
I suspect this is to do with the fact that `ExplicitModuleMap` is built by parsing out a JSON file (generated in the swift-driver), and at the point it is created perhaps `Identifier` keys are not available. @nkcsgexi can confirm.

Reply:
This is because we initially used `StringRef` to the underlying JSON buffer as the key, but we could potentially change it to `ASTContext`-owned identifiers now.